Sources/StreamChatUI/Gallery/GalleryVC.swift

```diff
@@ -400,12 +400,28 @@ open class GalleryVC:
     /// - Returns: An item to share.
     open func shareItem(at indexPath: IndexPath) -> Any? {
         guard let item = getItem(at: indexPath) else { return nil }
-        
-        if let image = item.attachment(payloadType: ImageAttachmentPayload.self) {
-            return image.imageURL
-        } else if let video = item.attachment(payloadType: VideoAttachmentPayload.self) {
-            return video.videoURL
-        } else {
+
+        switch item.type {
+        case .image:
+            let cell = attachmentsCollectionView
+                .cellForItem(at: indexPath) as? ImageAttachmentGalleryCell
+            return cell?.imageView.image
+        case .video:
+            guard let itemAttachment = item.attachment(payloadType: VideoAttachmentPayload.self),
+                  let urlData = try? Data(contentsOf: itemAttachment.videoURL) else {
+                return nil
+            }
+
+            let fileName = itemAttachment.id.messageId.lowercased()
+            let filePath = NSTemporaryDirectory().appending("\(fileName).mov")
+            let url = URL(fileURLWithPath: filePath)
+            do {
+                try urlData.write(to: url)
```

Comment:
@nuno-vieira 
This will cause incremental increase of the app size everytime the end user tries to share new video via our chat. We should make sure, to clean up the storage once the video is shared/eg when activity controller is dismissed.

Reply:
Here are the docs for `UIActivityViewController`s [completionWithItemsHandler](https://developer.apple.com/documentation/uikit/uiactivityviewcontroller/1622022-completionwithitemshandler).

Reply:
When the user closes the app the files will be removed, since we are using NSTemporaryDirectory. If you are talking about the memory while the app is being used, I think it does not make sense that if we keep sharing the same video multiple times that it will always take some time to process it but maybe this is a rare use case 🤔 What if we clean this only when we receive a memory warning ?